File: app/src/main/java/com/expansion/lg/kimaru/expansion/mzigos/Registration.java
package com.expansion.lg.kimaru.expansion.mzigos;

import java.util.Date;

/**
 * Created by kimaru on 3/11/17.
 */

public class Registration {

    String recruitment, name, phone, gender, district, subcounty, division, village, mark, langs, education;
    String id, occupation, comment, picture, country, chewName, chewNumber, ward, cuName, linkFacility;
    String otherTrainings, referralName, referralTitle, referralPhone, parish;
    Integer readEnglish, brac, bracChp, community, addedBy, proceed, synced;
    Long dob, dateMoved, dateAdded, noOfHouseholds, recruitmentTransportCost, transportCostToBranch;
    boolean chv, gokTrained, vht, accounts;
    int color = -1;
    Boolean read = false;

    public Registration(){

    }

    public Registration(String id, String mName, String mPhone, String mGender, String mDistrict,
                        String mSubcounty, String mDivision, String mVillage, String mMark,
                        String mLangs, String mEducation, String mOccupation, String mComment,
                        Long mDob, Integer mReadEnglish, String mRecruitment, String country,
                        Long mDateMoved, Integer mBrac, Integer mBracChp, Integer mCommunity,
                        Integer mAddedBy, Integer mProceed, Long mDateAdded, Integer mSynced,
                        String chewName, String chewNumber, String ward, String cuName,
                        String linkFacility, long noOfHouseholds, boolean isChv, boolean gokTrained,
                        String otherTrainings, String referralName, String referralTitle,
                        String referralPhone, boolean vht, boolean accounts, String parish,
                        Long recruitmentTransportCost, Long transportCostToBranch) {
        this.name = mName;
        this.phone = mPhone;
        this.recruitment = mRecruitment;
        this.gender = mGender;
        this.district = mDistrict;
        this.subcounty = mSubcounty;
        this.division = mDivision;
        this.village = mVillage;
        this.country = country;
        this.mark = mMark;
        this.langs = mLangs;
        this.education = mEducation;
        this.occupation = mOccupation;
        this.comment = mComment;
        this.dob = mDob;
        this.readEnglish = mReadEnglish;
        this.dateMoved = mDateMoved;
        this.brac = mBrac;
        this.bracChp = mBracChp;
        this.community = mCommunity;
        this.addedBy = mAddedBy;
        this.proceed = mProceed;
        this.dateAdded = mDateAdded;
        this.synced = mSynced;
        this.id = id;
        this.chewName = chewName;
        this.chewNumber = chewNumber;
        this.ward = ward;
        this.cuName = cuName;
        this.linkFacility = linkFacility;
        this.noOfHouseholds = noOfHouseholds;
        this.chv = isChv;
        this.gokTrained = gokTrained;
        this.otherTrainings = otherTrainings;
        this.referralName = referralName;
        this.referralTitle = referralTitle;
        this.referralPhone = referralPhone;
        this.vht = vht;
        this.accounts = accounts;
        this.parish = parish;
        this.recruitmentTransportCost = recruitmentTransportCost;
        this.transportCostToBranch = transportCostToBranch;
    }


    // Get Methods
    public String getName() {
        return name;
    }
    public String getPhone() {
        return phone;
    }
    public String getGender() {
        return gender;
    }
    public String getRecruitment() {
        return recruitment;
    }
    public String getDistrict() {
        return district;
    }
    public String getSubcounty() {
        return subcounty;
    }
    public String getDivision() {
        return division;
    }
    public String getVillage() {
        return village;
    }
    public String getMark() {
        return mark;
    }
    public String getLangs() {
        return langs;
    }
    public String getEducation() {
        return education;
    }
    public String getOccupation() {
        return occupation;
    }
    public String getComment() {
        return comment;
    }
    public Long getDob() {
        return dob;
    }
    public Integer getReadEnglish() {
        return readEnglish;
    }
    public Long getDateMoved() {
        return dateMoved;
    }
    public Integer getBrac() {
        return brac;
    }
    public Integer getBracChp() {
        return bracChp;
    }
    public Integer getCommunity() {
        return community;
    }
    public Integer getAddedBy() {
        return addedBy;
    }
    public Integer getProceed() {
        return proceed;
    }
    public Long getDateAdded() {
        return dateAdded;
    }
    public String getChewName() {
        return chewName;
    }
    public String getChewNumber() {
        return chewNumber;
    }
    public String getWard() {
        return ward;
    }
    public String getCuName() {
        return cuName;
    }
    public String getLinkFacility() {
        return linkFacility;
    }
    public boolean isChv() {
        return chv;
    }
    public boolean isGokTrained() {
        return gokTrained;
    }
    public Long getNoOfHouseholds() {
        return noOfHouseholds;
    }
    public String getOtherTrainings() {
        return otherTrainings;
    }

    public String getReferralName() {
        return referralName;
    }

    public String getReferralTitle() {
        return referralTitle;
    }

    public String getParish() {
        return parish;
    }

    public String getReferralPhone() {
        return referralPhone;
    }

    public boolean isAccounts() {
        return accounts;
    }

    public boolean isVht() {
        return vht;
    }

    public void setAccounts(boolean accounts) {
        this.accounts = accounts;
    }

    public void setParish(String parish) {
        this.parish = parish;
    }

    public void setReferralName(String referralName) {
        this.referralName = referralName;
    }

    public void setReferralPhone(String referralPhone) {
        this.referralPhone = referralPhone;
    }

    public void setReferralTitle(String referralTitle) {
        this.referralTitle = referralTitle;
    }

    public void setVht(boolean vht) {
        this.vht = vht;
    }

    public Long getRecruitmentTransportCost() {
        return recruitmentTransportCost;
    }

    public Long getTransportCostToBranch() {
        return transportCostToBranch;
    }

    // I added these methods
    public Integer getSynced() {
        return synced;
    }
    public String getId() {
        return id;
    }


    public String getPicture() {
        return "";
    }
    public int getColor() {
        return 1;
    }
    public boolean isImportant() {
        return true;
    }
    public boolean isRead() {
        return read;
    }
    public String getMessage(){return "This is a great Message";}

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    //Set Methods
    public void setName(String name) {
        this.name = name;
    }
    public void setPhone(String phone) {
        this.phone = phone;
    }
    public void setGender(String gender) {
        this.gender = gender;
    }
    public void setRecruitment(String recruitment) {
        this.recruitment = recruitment;
    }

    public void setDistrict(String district) {
        this.district = district;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public void setDivision(String division) {
        this.division = division;
    }

    public void setEducation(String education) {
        this.education = education;
    }

    public void setDob(Long dob) {
        this.dob = dob;
    }

    public void setLangs(String langs) {
        this.langs = langs;
    }

    public void setMark(String mark) {
        this.mark = mark;
    }

    public void setOccupation(String occupation) {
        this.occupation = occupation;
    }

    public void setSubcounty(String subcounty) {
        this.subcounty = subcounty;
    }

    public void setVillage(String village) {
        this.village = village;
    }

    public void setAddedBy(Integer addedBy) {
        this.addedBy = addedBy;
    }

    public void setBrac(Integer brac) {
        this.brac = brac;
    }

    public void setBracChp(Integer bracChp) {
        this.bracChp = bracChp;
    }

    public void setCommunity(Integer community) {
        this.community = community;
    }

    public void setDateAdded(Long dateAdded) {
        this.dateAdded = dateAdded;
    }

    public void setDateMoved(Long dateMoved) {
        this.dateMoved = dateMoved;
    }

    public void setProceed(Integer proceed) {
        this.proceed = proceed;
    }

    public void setReadEnglish(Integer readEnglish) {
        this.readEnglish = readEnglish;
    }

    public void setSynced(Integer synced) {
        this.synced = synced;
    }
    public void setPicture(String picture){ this.picture = picture;}

    public void setColor(int color) {
        this.color = color;
    }

    public void setRead(boolean read) {
        this.read = read;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Boolean getRead() {
        return read;
    }

    public void setChewName(String chewName) {
        this.chewName = chewName;
    }

    public void setChewNumber(String chewNumber) {
        this.chewNumber = chewNumber;
    }

    public void setChv(boolean chv) {
        this.chv = chv;
    }

    public void setCuName(String cuName) {
        this.cuName = cuName;
    }

    public void setGokTrained(boolean gokTrained) {
        this.gokTrained = gokTrained;
    }

    public void setLinkFacility(String linkFacility) {
        this.linkFacility = linkFacility;
    }

    public void setNoOfHouseholds(Long noOfHouseholds) {
        this.noOfHouseholds = noOfHouseholds;
    }

    public void setOtherTrainings(String otherTrainings) {
        this.otherTrainings = otherTrainings;
    }

    public void setRead(Boolean read) {
        this.read = read;
    }

    public void setWard(String ward) {
        this.ward = ward;
    }

    public Long getAge(){
        Long currentDate =  new Date().getTime();
        Long epochYear = 31556926000L;
        Long yearEpoch = currentDate - this.dob;
        Long ageYear = yearEpoch / epochYear;
        return ageYear;
    }

    public void setRecruitmentTransportCost(Long recruitmentTransportCost) {
        this.recruitmentTransportCost = recruitmentTransportCost;
    }

    public void setTransportCostToBranch(Long transportCostToBranch) {
        this.transportCostToBranch = transportCostToBranch;
    }

    public boolean hasPassed(){
        Long age = this.getAge();
        if ( 30 < age || age > 55){
            return false;
        }else if (this.readEnglish == 0) {
            return false;
        }else if (this.dateMoved < 2) {
            return false;
        }else if (brac == 1 && bracChp == 1){
            return false;
        }else if (Integer.valueOf(this.education) < 7 || Integer.valueOf(this.education) > 12){
            return false;
        }else {
            return true;
        }
    }

}

File: app/src/main/java/com/expansion/lg/kimaru/expansion/mzigos/Registration.kt
package com.expansion.lg.kimaru.expansion.mzigos

import java.util.Date

/**
 * Created by kimaru on 3/11/17.
 */

class Registration {

    lateinit var recruitment: String
    // Get Methods
    //Set Methods
    lateinit var name: String
    lateinit var phone: String
    lateinit var gender: String
    lateinit var district: String
    lateinit var subcounty: String
    lateinit var division: String
    lateinit var village: String
    lateinit var mark: String
    lateinit var langs: String
    lateinit var education: String
    lateinit var id: String
    lateinit var occupation: String
    lateinit var comment: String
    lateinit internal var picture: String
    lateinit var country: String
    lateinit var chewName: String
    lateinit var chewNumber: String
    lateinit var ward: String
    lateinit var cuName: String
    lateinit var linkFacility: String
    lateinit var otherTrainings: String
    lateinit var referralName: String
    lateinit var referralTitle: String
    lateinit var referralPhone: String
    lateinit var parish: String
    lateinit var chewUuid: String
    lateinit var maritalStatus: String
    lateinit var subCountyId: String
    lateinit var parishId: String
    lateinit var villageId: String
    var readEnglish: Int? = null
    var brac: Int? = null
    var bracChp: Int? = null
    var community: Int? = null
    var addedBy: Int? = null
    var proceed: Int? = null
    // I added these methods
    var synced: Int? = null
    var dob: Long? = null
    var dateMoved: Long? = null
    var dateAdded: Long? = null
    var noOfHouseholds: Long? = null
    var recruitmentTransportCost: Long? = null
    var transportCostToBranch: Long? = null
    var isChv: Boolean = false
    var isGokTrained: Boolean = false
    var isVht: Boolean = false
    var isAccounts: Boolean = false
    internal var color = -1
    internal var read: Boolean? = false
    val isImportant: Boolean
        get() = true
    val isRead: Boolean
        get() = read!!
    val message: String
        get() = "This is a great Message"

    val age: Long?
        get() {
            val currentDate = Date().time
            val epochYear = 31556926000L
            val yearEpoch = currentDate - this.dob!!
            return yearEpoch / epochYear
        }

    constructor() {

    }

    constructor(id: String, mName: String, mPhone: String, mGender: String, mDistrict: String,
                mSubcounty: String, mDivision: String, mVillage: String, mMark: String,
                mLangs: String, mEducation: String, mOccupation: String, mComment: String,
                mDob: Long?, mReadEnglish: Int?, mRecruitment: String, country: String,
                mDateMoved: Long?, mBrac: Int?, mBracChp: Int?, mCommunity: Int?,
                mAddedBy: Int?, mProceed: Int?, mDateAdded: Long?, mSynced: Int?,
                chewName: String, chewNumber: String, ward: String, cuName: String,
                linkFacility: String, noOfHouseholds: Long, isChv: Boolean, gokTrained: Boolean,
                otherTrainings: String, referralName: String, referralTitle: String,
                referralPhone: String, vht: Boolean, accounts: Boolean, parish: String,
                recruitmentTransportCost: Long?, transportCostToBranch: Long?, chewUuid: String,
                maritalStatus: String, subCountyId: String, parishId: String, villageId: String) {
        this.name = mName
        this.phone = mPhone
        this.recruitment = mRecruitment
        this.gender = mGender
        this.district = mDistrict
        this.subcounty = mSubcounty
        this.subCountyId = subCountyId
        this.division = mDivision
        this.village = mVillage
        this.country = country
        this.mark = mMark
        this.langs = mLangs
        this.education = mEducation
        this.occupation = mOccupation
        this.comment = mComment
        this.dob = mDob
        this.readEnglish = mReadEnglish
        this.dateMoved = mDateMoved
        this.brac = mBrac
        this.bracChp = mBracChp
        this.community = mCommunity
        this.addedBy = mAddedBy
        this.proceed = mProceed
        this.dateAdded = mDateAdded
        this.synced = mSynced
        this.id = id
        this.chewName = chewName
        this.chewNumber = chewNumber
        this.ward = ward
        this.cuName = cuName
        this.linkFacility = linkFacility
        this.noOfHouseholds = noOfHouseholds
        this.isChv = isChv
        this.isGokTrained = gokTrained
        this.otherTrainings = otherTrainings
        this.referralName = referralName
        this.referralTitle = referralTitle
        this.referralPhone = referralPhone
        this.isVht = vht
        this.isAccounts = accounts
        this.parish = parish
        this.recruitmentTransportCost = recruitmentTransportCost
        this.transportCostToBranch = transportCostToBranch
        this.chewUuid = chewUuid
        this.maritalStatus = maritalStatus
        this.subCountyId = subCountyId
        this.villageId = villageId
        this.parishId = parishId
    }

    fun getPicture(): String {
        return ""
    }

    fun getColor(): Int {
        return 1
    }

    fun setPicture(picture: String) {
        this.picture = picture
    }

    fun setColor(color: Int) {
        this.color = color
    }

    fun setRead(read: Boolean) {
        this.read = read
    }

    fun getRead(): Boolean? {
        return read
    }

    fun setRead(read: Boolean?) {
        this.read = read
    }

    //    public boolean hasPassed(){
    //        Integer age = getAge().intValue();
    //        if ( age < 30 || age > 55){
    //            return false;
    //        }else if (this.readEnglish.equals(0)) {
    //            return false;
    //        }else if (this.dateMoved < 2) {
    //            return false;
    //        }else if (brac.equals(1) && bracChp.equals(1)){
    //            return false;
    //        }else if (this.country.equalsIgnoreCase("UG")){
    //            // if the education is less than P7
    //            if (Integer.valueOf(this.education) < 2 || Integer.valueOf(this.education) > 7 ){
    //                return false;
    //            }else {
    //                return true;
    //            }
    //
    //        }else {
    //            if (Integer.valueOf(this.education) < 10 || Integer.valueOf(this.education) > 14 ){
    //                return false;
    //            }else {
    //                return true;
    //            }
    //        }
    //    }

    fun hasPassed(): Boolean {
        val age = age!!.toInt()
        return if (age < 30 || age > 55) {
            false
        } else if (this.readEnglish == 0) {
            false
        } else if (this.dateMoved!! < 2) {
            false
        } else if (this.bracChp == 1) {
            false
        } else if (this.country.equals("UG", ignoreCase = true)) {
            // if the education is less than P7
            !(Integer.valueOf(this.education) < 2 || Integer.valueOf(this.education) > 7)
        } else {
            !(Integer.valueOf(this.education) < 10 || Integer.valueOf(this.education) > 14)
        }
    }

}